solutions/cses-1202.mdx

```diff
@@ -5,86 +5,99 @@ title: Investigation
+distance and reset $\texttt{num}[u]$, $\texttt{minf}[u]$, and $\texttt{maxf}[u]$.
+
+We also have to take into
 consideration if we can reach $u$ in an equivalent distance. If so, we update:
 
 $$
-num[u] = num[v] + num[u]
+\texttt{num}[u] = \texttt{num}[v] + \texttt{num}[u]
 $$
 
 $$
-minf[u] = min(minf[u], minf[v] + 1)
+\texttt{minf}[u] = \min(\texttt{minf}[u], \texttt{minf}[v] + 1)
 $$
 
 $$
-maxf[u] = max(maxf[u], maxf[v] + 1)
+\texttt{maxf}[u] = \max(\texttt{maxf}[u], \texttt{maxf}[v] + 1)
 $$
 
+## Implementation
+
+**Time Complexity:** $\mathcal{O}((N + M)\log N)$
+
+<LanguageSection>
+	
+<CPPSection>
+
 ```cpp
 #include <bits/stdc++.h>
 using namespace std;
 
 typedef long long ll;
-vector<pair<ll, int>> edge[100001];
-ll dist[100001]; ll num[100001];
-int minf[100001];int maxf[100001];
-bool v[100001];
-ll MAX = 1000000000000000L; ll MOD = 1000000007L;
-void djikstra(int s){
+
+const int MAXN = 100001;
+const ll MAX = 0x3f3f3f3f3f3f3f3f;
+const int MOD = 1000000007;
+
+vector<pair<ll, int>> edge[MAXN];
+ll dist[MAXN], num[MAXN];
+int minf[MAXN], maxf[MAXN];
+bool v[MAXN];
+
+void djikstra(int s) {
 	priority_queue<pair<ll, int>, vector<pair<ll, int>>, greater<pair<ll, int>>> pq;
-	pair<ll, int> p; p.first = 0; p.second = s;
-	pq.push(p);
-	num[s] = 1; dist[s] = 0;
-	while(!pq.empty()){
-		pair<ll, int> cur = pq.top(); pq.pop();
-		if(!v[cur.second]){
-			v[cur.second] = true;
-			for(pair<ll, int> e : edge[cur.second]){
-				ll cost = e.first; int next = e.second;
-				ll alt = cost + dist[cur.second];
-				if(alt == dist[next]){
-					num[next] = (num[next] + num[cur.second]) % MOD;
-					minf[next] = min(minf[next], minf[cur.second]+1);
-					maxf[next] = max(maxf[next], maxf[cur.second]+1);
-					e.first = dist[next];
-					pq.push(e);
-				}
-				else if(alt < dist[next]){
-					dist[next] = alt;
-					num[next] = num[cur.second];
-					minf[next] = minf[cur.second]+1;
-					maxf[next] = maxf[cur.second]+1;
-					e.first = dist[next];
-					pq.push(e);
+	pq.push({dist[s] = 0, s});
+	num[s] = 1;
+	while (!pq.empty()) {
+		auto [cost, vert] = pq.top(); pq.pop();
+		if (!v[vert]) {
+			v[vert] = true;
+			for (auto [cost, next] : edge[vert]) {
+				ll alt = cost + dist[vert];
+				if (alt == dist[next]) {
+					num[next] = (num[next] + num[vert]) % MOD;
+					minf[next] = min(minf[next], minf[vert] + 1);
+					maxf[next] = max(maxf[next], maxf[vert] + 1);
+				} else if(alt < dist[next]) {
+					num[next] = num[vert];
+					minf[next] = minf[vert] + 1;
+					maxf[next] = maxf[vert] + 1;
+					pq.push({dist[next] = alt, next});
 				}
 			}
 		}
 	}
 }
-int main(){
+int main() {
 	ios_base::sync_with_stdio(0); cin.tie(0);
-	int n; int m; cin >> n >> m;
-	for(int i = 0; i < m; i++){
-		int a,b,c; cin >> a >> b >> c;
-		pair<ll, int> d; d.first = c; d.second = b;
-		edge[a].push_back(d);
+	int n, m; cin >> n >> m;
+	for (int i = 0; i < m; i++) {
+		int a, b, c; cin >> a >> b >> c;
```

Comment:
actual variable names would be nice

Reply:
these are the names given in the problem statement so it's fine

Reply:
![image](https://user-images.githubusercontent.com/55369003/126718273-33d2c4e9-63e7-419d-a87d-98ae878d7274.png)
i mean dong believes in these changes himself so